System/Posix/Fcntl.hsc

```diff
@@ -101,3 +112,68 @@ foreign import capi safe "fcntl.h posix_fallocate"
 fileAllocate _ _ _ = ioError (ioeSetLocation unsupportedOperation
                               "fileAllocate")
 #endif
+
+-- -----------------------------------------------------------------------------
+-- File caching
+
+-- | Performs the @fcntl(2)@ operation on a file-desciptor to get the cache mode.
+--
+-- If the cache mode is 'False', then cache effects for file system reads and
+-- writes are minimised or otherwise eliminated. If the cache mode is 'True',
+-- then cache effects occur like normal.
```

Comment:
I now realise this is actually describing the reverse of what it is doing :facepalm:

Reply:
I could rename the functions to `fileSetNoCaching`/`fileGetNoCaching`, or invert the interpretation of the boolean argument. Preferences?

Reply:
`fileSetNoCaching` feels like `unless`, which always causes me brain  aneurysm.